File: packages/pi-plugin/src/__tests__/e2e/fixtures/sample.ts
import { readFileSync } from "node:fs";
import { join } from "node:path";

export const DEFAULT_SUFFIX = "!";
const LOCAL_SEPARATOR = ":";

function normalize(input: string): string {
  return input.trim().toUpperCase();
}

function decorate(input: string): string {
  return `[${input}]`;
}

export function funcA(input: string): string {
  return `A${LOCAL_SEPARATOR}${input}`;
}

export function funcB(name: string): string {
  return `${normalize(name)}${DEFAULT_SUFFIX}`;
}

export function funcC(parts: string[]): string {
  return decorate(parts.join(join("", "-")));
}

export class SampleService {
  greet(name: string): string {
    return funcB(name);
  }

  readFirstLine(filePath: string): string {
    return readFileSync(filePath, "utf8").split("\n")[0] ?? "";
  }
}

File: packages/pi-plugin/src/__tests__/e2e/fixtures/sample.py
DEFAULT_NAME = "world"
MAX_COUNT = 3


def greet(name: str) -> str:
    return f"Hello, {name}"


def combine(parts: list[str]) -> str:
    return "-".join(parts)


class SampleTool:
    def upper(self, value: str) -> str:
        return value.upper()

    def lower(self, value: str) -> str:
        return value.lower()
